Content.Client/IconSmoothing/IconSmoothSystem.cs

```diff
@@ -7,494 +7,496 @@
 using Robust.Shared.Map.Enumerators;
 using static Robust.Client.GameObjects.SpriteComponent;
 
-namespace Content.Client.IconSmoothing
+namespace Content.Client.IconSmoothing;
+
+// TODO: just make this set appearance data?
+/// <summary>
+///     Entity system implementing the logic for <see cref="IconSmoothComponent"/>
+/// </summary>
+[UsedImplicitly]
+public sealed partial class IconSmoothSystem : EntitySystem
 {
-    // TODO: just make this set appearance data?
-    /// <summary>
-    ///     Entity system implementing the logic for <see cref="IconSmoothComponent"/>
-    /// </summary>
-    [UsedImplicitly]
-    public sealed partial class IconSmoothSystem : EntitySystem
+    [Dependency] private readonly MapSystem _map = default!;
+
+    private readonly Queue<EntityUid> _dirtyEntities = new();
+    private readonly Queue<EntityUid> _anchorChangedEntities = new();
+
+    private int _generation;
+
+    public void SetEnabled(EntityUid uid, bool value, IconSmoothComponent? component = null)
     {
-        private readonly Queue<EntityUid> _dirtyEntities = new();
-        private readonly Queue<EntityUid> _anchorChangedEntities = new();
+        if (!Resolve(uid, ref component, false) || value == component.Enabled)
+            return;
 
-        private int _generation;
+        component.Enabled = value;
+        DirtyNeighbours(uid, component);
+    }
+
+    public override void Initialize()
+    {
+        base.Initialize();
 
-        public void SetEnabled(EntityUid uid, bool value, IconSmoothComponent? component = null)
+        InitializeEdge();
+        SubscribeLocalEvent<IconSmoothComponent, AnchorStateChangedEvent>(OnAnchorChanged);
+        SubscribeLocalEvent<IconSmoothComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<IconSmoothComponent, ComponentStartup>(OnStartup);
+    }
+
+    private void OnStartup(EntityUid uid, IconSmoothComponent component, ComponentStartup args)
+    {
+        var xform = Transform(uid);
+        if (xform.Anchored)
         {
-            if (!Resolve(uid, ref component, false) || value == component.Enabled)
-                return;
+            component.LastPosition = TryComp<MapGridComponent>(xform.GridUid, out var grid)
+                ? (xform.GridUid.Value, _map.TileIndicesFor(xform.ParentUid, grid, xform.Coordinates))
```

Comment:
Can you use `xform.GridUid` instead of `xform.ParentUid` just in case?

Same for all the other usages.

Reply:
Done. I had to still use ParentUid in one place because apparently entities that are just getting deleted still have their transformcomponent but return null on GridUid. Their ParentId seems to be a dummy, but things still work with that (and is how it has been until now)